lib/Runtime/PlatformAgnostic/Platform/Common/Trace.cpp

```diff
@@ -0,0 +1,320 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Trace light
+// Standalone solution to trace function calls in a *nix application
+
+// Notes to Embedder;
+
+// This file is originally implemented for ChakraCore.
+// However it is not dependent to any part of the project (and supposed to keep it that way)
+// To use with your project, simple copy-paste of this file is sufficient.
+// Required compiler args are -finstrument-functions
+// -finstrument-functions enables GCC/Clang to inject __cyg_profile_func_enter/exit
+// As of 3.9, clang doesn't support finstrument_functions_exclude_file_list or similar (reason for IN_TRACE)
+// This version doesn't support multithread profiling. Records the entries from initial thread only
+
+// In case of bug-fix and/or improvements, please contribute back to https://github.com/Microsoft/ChakraCore
+
+// Dealing with output;
+
+// Converting from Address
+// OSX ->    atos -o <executable folder> <address here>
+// Linux ->  addr2line -e <executable folder> <address here>
+
+// gdb -> info symbol <address here>
+
+// DEFINITIONS
+
+// #define TRACE_OUTPUT_TARGET_FILE // <- uncomment this if you want the trace is written to a file (TraceOutput.txt)
+// #define TRACE_OUTPUT_TO_LOGCAT // <- uncomment this if you want to target Logcat for Android instead of stderr
+
+// #define ENABLE_CC_XPLAT_TRACE // uncomment this to use this code with your project or add ENABLE_CC_XPLAT_TRACE to your compile definitions
+#ifdef ENABLE_CC_XPLAT_TRACE
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <string.h>
+#include <string>
+#define _GNU_SOURCE
+#include <dlfcn.h> // dladdr
+
+#if defined(__APPLE__)
+#include <mach-o/dyld.h> // _NSGetExecutablePath
+#elif defined(__linux__)
+#include <unistd.h> // readlink
+#endif
+
+
+#if defined(__ANDROID__) && defined(TRACE_OUTPUT_TO_LOGCAT)
+#include <android/log.h>
+#define PRINT_ERROR(...) \
+    __android_log_print(ANDROID_LOG_ERROR, "TRACE_LIGHT", __VA_ARGS__)
+#else
+#define PRINT_ERROR(...) \
+    fprintf(stderr, __VA_ARGS__);
+#endif
+
+#define TRACE_VERSION_STRING "Trace light v0.01"
+#define FILE_BUFFER_SIZE 16384
+#define SELF_PATH_SIZE    2048
+static int SELF_PATH_LENGTH = 0;
+static FILE *FTRACE = NULL;
+static _Thread_local size_t LAST_TICK = -1;
+static bool IN_TRACE = 0;
+static unsigned long CALL_IN_COUNTER = 0;
+static char SELF_PATH[SELF_PATH_SIZE];
+static char FILE_BUFFER[FILE_BUFFER_SIZE];
+static int BUFFER_POS = 0;
+
+__attribute__((no_instrument_function))
+static void
+GetBinaryLocation()
```

Comment:
Yes, I was aware of that. Questionable though if it is worth duplicating code for that goal.

Also, is this only useful for diagnostics builds and can be excluded form "release"?